aicsimageio/readers/czi_reader.py

```diff
@@ -0,0 +1,784 @@
+
+    @staticmethod
+    def _dims_shape_to_scene_dims_shape(
+        dims_shape: List[Dict], scene_index: int, consistent: bool
+    ) -> Dict:
+        dims_shape_index = 0 if consistent else scene_index
+        dims_shape_dict = dims_shape[dims_shape_index]
+        dims_shape_dict.pop("S", None)
+        return dims_shape_dict
+
+    @staticmethod
+    def _stitch_tiles(
+        data: types.ArrayLike,
+        data_dims_shape: Dict,
+        bboxes: Dict,
+        mosaic_bbox: BBox,
+    ) -> types.ArrayLike:
+        # Assumptions: 1) docs for ZEISSRAW(CZI) say:
+        #   Scene – for clustering items in X/Y direction (data belonging to
+        #   contiguous regions of interests in a mosaic image).
+
+        # 'S' in czi is Scene not Samples, 'A' is sAmples
+
+        # Create empty array
+        arr_shape_list = []
+
+        ordered_dims_present = [
+            dim for dim in DEFAULT_CZI_DIMENSION_ORDER_LIST if dim in data_dims_shape
+        ]
+        for dim in ordered_dims_present:
+            if dim not in REQUIRED_CZI_CHUNK_DIMS:
+                arr_shape_list.append(data_dims_shape[dim][1])
+            if dim is DimensionNames.SpatialY:
+                arr_shape_list.append(mosaic_bbox.h)
+            if dim is DimensionNames.SpatialX:
+                arr_shape_list.append(mosaic_bbox.w)
+            if dim == "A":
+                arr_shape_list.append(data_dims_shape[dim][1])
+
+        ans = None
+        if type(data) is da.Array:
+            ans = da.zeros(
+                arr_shape_list,
+                chunks=data.chunks,  # type: ignore
+                dtype=data.dtype,
+            )
+        else:
+            ans = np.zeros(arr_shape_list, dtype=data.dtype)
+
+        for (tile_info, box) in bboxes.items():
+            # construct data indexes to use
+            tile_dims = tile_info.dimension_coordinates
+            tile_dims.pop("S", None)
+            tile_dims.pop("B", None)
+            data_indexes = [
+                tile_dims[t_dim]
+                for t_dim in DEFAULT_CZI_DIMENSION_ORDER_LIST_WITH_MOSAIC_TILES
+                if t_dim in tile_dims.keys() and t_dim not in REQUIRED_CZI_CHUNK_DIMS
+            ]
+            # add Y and X
+            data_indexes.append(slice(None))  # Y ":"
+            data_indexes.append(slice(None))  # X ":"
+            # if "A" in tile_dims.keys():
+            #     data_indexes.append(slice(None))
+
+            # construct data indexes for ans
+            ans_indexes = []
+            for dim in ordered_dims_present:  # DEFAULT_CZI_DIMENSION_ORDER_LIST:
+                if dim not in [
+                    "S",
+                    DimensionNames.MosaicTile,
+                    DimensionNames.SpatialY,
+                    DimensionNames.SpatialX,
+                ]:
+                    if dim in tile_dims.keys():
+                        ans_indexes.append(tile_dims[dim])
+                if dim is DimensionNames.SpatialY:
```

Comment:
Super nitpicky, but can you add an empty line between this if and the above if statement. Just more visual distinction that these are in different scopes. (its also 10am so maybe I'm weird)